bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -127,6 +127,21 @@ public GarbageCollectorThread(ServerConfiguration conf, LedgerManager ledgerMana
                 Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("GarbageCollectorThread")));
     }
 
+    private EntryLogMetadataMap createEntryLogMetadataMap(boolean gcPersistentEntrylogMetadataMapEnabled) {
+        final String baseDir = this.conf.getGcPersistentEntrylogMetadataMapPath() != null
+                ? this.conf.getGcPersistentEntrylogMetadataMapPath()
+                : this.entryLogger.getLedgerDirsManager().getAllLedgerDirs().get(0).toString();
```

Comment:
since you are adding new persistence state to the bookie - 'PersistentEntrylogMetadataMapPath', have you thought through all the persistence scenarios? like any changes required to cookie, what it means for bookie formatting, backward compatibility / forward compatibility scenarios,...

Reply:
I think bookie-formatting is the only operation that requires metadata cleanup for which I made the change. also, GC has fall-back mechanism to use InMemoryMap in case of PersistentMap is not accessible and PersistentMap is recoverable by extracting entryLog so, that should take care of forward/backward compatibility concerns.